packetbeat/protos/amqp/amqp_fields.go

```diff
@@ -0,0 +1,199 @@
+package amqp
+
+import (
+	"encoding/binary"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func getTable(fields common.MapStr, data []byte, offset uint32) (next uint32, err bool, exists bool) {
+	ret := make(common.MapStr)
+
+	length := binary.BigEndian.Uint32(data[offset : offset+4])
+
+	// size declared too big
+	if length > uint32(len(data[offset+4:])) {
+		return 0, true, false
+	}
+	if length > 0 {
+		exists = true
+		err := fieldUnmarshal(ret, data[offset+4:offset+4+length], 0, length, -1)
+		if err {
+			logp.Warn("Error while parsing a field table")
+			return 0, true, false
+		}
+		if fields == nil {
+			fields = ret
+		} else {
+			fields.Update(ret)
+		}
+	}
+	return length + 4 + offset, false, exists
+}
+
+func getArray(fields common.MapStr, data []byte, offset uint32) (next uint32, err bool, exists bool) {
```

Comment:
It's the same difference as a hash and an array in Ruby, for example. In a table, you will have several named members with different types, for exemple 
`amqp.mytable.greetings = "hello"`

`amqp.mytable.answer = 42`

`amqp.mytable.mynumber = 28.5`

An array will have several members of the same type with an index

`amqp.myarray.0 = "hello"`

`amqp.myarray.1 = "hallo"`

`amqp.myarray.2 = "salut"`

The same function is used to parse them both, except that instead of getting the name of the parameter in getArray, I maintain a counter of the index starting from 0.